### PR TITLE
Demo and API docs improvements

### PR DESCRIPTION
**Pull request by web-padawan**
Connected to #133 

The PR is to track overall progress, better to review on the per-commit basis. The docs are to be rebuilt at the very end once we agree on the set of changes.

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-router/136)

### CODE REVIEW

**Comment by vlukashov** [comment]
Review status: 0 of 19 files reviewed at latest revision, all discussions resolved.

---

*[demo/vaadin-router-lazy-loading-demos.html, line 79 at r1](https://reviewable.io/reviews/vaadin/vaadin-router/136#-LEFBNCP97WhSIoBMG-b:-LEFBNCP97WhSIoBMG-c:b-u2seo3) ([raw file](https://github.com/vaadin/vaadin-router/blob/2dd9e10646369424f01d20fe3bf2fc9ede8e5122/demo/vaadin-router-lazy-loading-demos.html#L79)):*
> ```HTML
>               Polymer.importHref(
>                 `${window.Vaadin.demoComponentsRoot}/user.bundle.html`, () => {
>                   resolve(context.component(context.route.component));
> ```

could be simplified to `() => resolve()`

---

*[demo/demo-elements/user.bundle.html, line 22 at r1](https://reviewable.io/reviews/vaadin/vaadin-router/136#-LEFB7rl1oBW8AD42m-g:-LEFB7rl1oBW8AD42m-h:bjcthdh) ([raw file](https://github.com/vaadin/vaadin-router/blob/2dd9e10646369424f01d20fe3bf2fc9ede8e5122/demo/demo-elements/user.bundle.html#L22)):*
> ```HTML
>     }
> 
>     customElements.define(XUSerHtmlBundleView.is, XUSerHtmlBundleView);
> ```

`XUSerHtmlBundleView` looks like a typo, should be `XUserHtmlBundleView` (`User`, not `USer`)

---

*[demo/demo-elements/user.bundle.js, line 2 at r1](https://reviewable.io/reviews/vaadin/vaadin-router/136#-LEFBIWW41SWX8ufMzHZ:-LEFBIWW41SWX8ufMzH_:b-tpcfap) ([raw file](https://github.com/vaadin/vaadin-router/blob/2dd9e10646369424f01d20fe3bf2fc9ede8e5122/demo/demo-elements/user.bundle.js#L2)):*
> ```JavaScript
> (() => {
>   class XUSerJsBundleView extends Polymer.Element {
> ```

`User`, not `USer`?

---


*Comments from [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-router/136)*

**Comment by web-padawan** [comment]
Review status: 0 of 19 files reviewed at latest revision, 3 unresolved discussions.

---

*[demo/vaadin-router-lazy-loading-demos.html, line 79 at r1](https://reviewable.io/reviews/vaadin/vaadin-router/136#-LEFBNCP97WhSIoBMG-b:-LEFYoGV8FlMv609FWwX:b-896fix) ([raw file](https://github.com/vaadin/vaadin-router/blob/2dd9e10646369424f01d20fe3bf2fc9ede8e5122/demo/vaadin-router-lazy-loading-demos.html#L79)):*
<details><summary><i>Previously, vlukashov (Viktor Lukashov) wrote…</i></summary><blockquote>

could be simplified to `() => resolve()`
</blockquote></details>

Done.

---

*[demo/demo-elements/user.bundle.html, line 22 at r1](https://reviewable.io/reviews/vaadin/vaadin-router/136#-LEFB7rl1oBW8AD42m-g:-LEFYpiUA03sBznYbIBS:b-896fix) ([raw file](https://github.com/vaadin/vaadin-router/blob/2dd9e10646369424f01d20fe3bf2fc9ede8e5122/demo/demo-elements/user.bundle.html#L22)):*
<details><summary><i>Previously, vlukashov (Viktor Lukashov) wrote…</i></summary><blockquote>

`XUSerHtmlBundleView` looks like a typo, should be `XUserHtmlBundleView` (`User`, not `USer`)
</blockquote></details>

Done.

---

*[demo/demo-elements/user.bundle.js, line 2 at r1](https://reviewable.io/reviews/vaadin/vaadin-router/136#-LEFBIWW41SWX8ufMzHZ:-LEFYqPb1LTHomDy9LWQ:b-896fix) ([raw file](https://github.com/vaadin/vaadin-router/blob/2dd9e10646369424f01d20fe3bf2fc9ede8e5122/demo/demo-elements/user.bundle.js#L2)):*
<details><summary><i>Previously, vlukashov (Viktor Lukashov) wrote…</i></summary><blockquote>

`User`, not `USer`?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-router/136)*

**Comment by SomeoneToIgnore** [comment]
*[src/utils.js, line 20 at r13](https://reviewable.io/reviews/vaadin/vaadin-router/136#-LEIvn3PENE96UaELBbE:-LEIvn3PENE96UaELBbF:b-hm5p17) ([raw file](https://github.com/vaadin/vaadin-router/blob/a445c020a99bee60f4979cd4338b4bd3c05832d3/src/utils.js#L20)):*
Not a blocker: we might rephrase this to explicitly state that 'action' should be a function, since the user may put something different here and wonder why did he get the warning.

For instance: 

> or "action" function but none found.

---


*Comments from [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-router/136)*

**Comment by vlukashov** [comment]
Review status: 0 of 19 files reviewed at latest revision, all discussions resolved.

---

*[src/utils.js, line 18 at r3](https://reviewable.io/reviews/vaadin/vaadin-router/136#-LEIw8ggCOHub3XKMeEO:-LEIw8ggCOHub3XKMeEP:b-xm3ybg) ([raw file](https://github.com/vaadin/vaadin-router/blob/980fe8168e79e6e3f67c1269385e8ff1578a533f/src/utils.js#L18)):*
> ```JavaScript
> 
>   const stringKeys = ['component', 'redirect', 'bundle'];
>   if (typeof route.action !== 'function' && !Array.isArray(route.children) && !stringKeys.some(key => typeof route[key] === 'string')) {
> ```

Is the `inactivate` property omitted intentionally?

---


*Comments from [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-router/136)*

**Comment by web-padawan** [comment]
Review status: 17 of 19 files reviewed at latest revision, 3 unresolved discussions.

---

*[src/router.js, line 182 at r5](https://reviewable.io/reviews/vaadin/vaadin-router/136#-LEIx61RESDBYSdBv-jH:-LEJ-zXs9HeXqyBtGAP2:b-896fix) ([raw file](https://github.com/vaadin/vaadin-router/blob/87891e3454cf2f89d7216208a5a24c163af6e7c4/src/router.js#L182)):*
<details><summary><i>Previously, vlukashov (Viktor Lukashov) wrote…</i></summary><blockquote>

a closing back tick is missing: `context`
</blockquote></details>

Done.

---

*[src/router.js, line 183 at r5](https://reviewable.io/reviews/vaadin/vaadin-router/136#-LEIynIj4XXj6bDzRL3m:-LEJ0-C23oD0jVNDqabg:b-896fix) ([raw file](https://github.com/vaadin/vaadin-router/blob/87891e3454cf2f89d7216208a5a24c163af6e7c4/src/router.js#L183)):*
<details><summary><i>Previously, vlukashov (Viktor Lukashov) wrote…</i></summary><blockquote>

Would be nice to have a link to the 'Route Actions' live demos here.
</blockquote></details>

Done.

---

*[src/utils.js, line 18 at r3](https://reviewable.io/reviews/vaadin/vaadin-router/136#-LEIw8ggCOHub3XKMeEO:-LEJ09dX89hzLB3GMwQ2:b-twlzam) ([raw file](https://github.com/vaadin/vaadin-router/blob/980fe8168e79e6e3f67c1269385e8ff1578a533f/src/utils.js#L18)):*
<details><summary><i>Previously, vlukashov (Viktor Lukashov) wrote…</i></summary><blockquote>

Is the `inactivate` property omitted intentionally?
</blockquote></details>

Yes, we probably do not expect the "inactivate" to be the only meaningful property. We can reconsider this in further PRs, though.

---

*[src/utils.js, line 20 at r13](https://reviewable.io/reviews/vaadin/vaadin-router/136#-LEIvn3PENE96UaELBbE:-LEJ07XA42zGMLeoQDqv:b-896fix) ([raw file](https://github.com/vaadin/vaadin-router/blob/a445c020a99bee60f4979cd4338b4bd3c05832d3/src/utils.js#L20)):*
<details><summary><i>Previously, SomeoneToIgnore (Kirill Bulatov) wrote…</i></summary><blockquote>

Not a blocker: we might rephrase this to explicitly state that 'action' should be a function, since the user may put something different here and wonder why did he get the warning.

For instance: 

> or "action" function but none found.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-router/136)*

**Comment by vlukashov** [comment]
Review status: 16 of 19 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/router.js, line 182 at r5](https://reviewable.io/reviews/vaadin/vaadin-router/136#-LEJ6UGL27knqMRTJIZl:-LEJ6UGMEmuXWAXfM_B7:bqf3n3b) ([raw file](https://github.com/vaadin/vaadin-router/blob/87891e3454cf2f89d7216208a5a24c163af6e7c4/src/router.js#L182)):*
> ```JavaScript
>    * If present, this function is always invoked first, disregarding of the other properties' presence.
>    * If the action returns a non-empty result, current route resolution is finished and other route config properties are ignored.
>    * `context parameter can be used for asynchronously getting the resolved route contents via `context.next()
> ```

another closing back tick is missing

---


*Comments from [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-router/136)*

**Comment by vlukashov** [comment]
Review status: 16 of 19 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/router.js, line 30 at r8](https://reviewable.io/reviews/vaadin/vaadin-router/136#-LEJ72IQBaYqW7IZ3nyC:-LEJ72IQBaYqW7IZ3nyD:b-v8vz99) ([raw file](https://github.com/vaadin/vaadin-router/blob/0f2435fb7cc69312f63cf60e669e68e26510be5f/src/router.js#L30)):*
> ```
> Use the `new Router(outlet)`
> ```

Use ~the~ `new Router(outlet)`

---

*[src/router.js, line 41 at r8](https://reviewable.io/reviews/vaadin/vaadin-router/136#-LEJ7DyA9sAUkHhaijOV:-LEJ7DyA9sAUkHhaijOW:b5gptp6) ([raw file](https://github.com/vaadin/vaadin-router/blob/0f2435fb7cc69312f63cf60e669e68e26510be5f/src/router.js#L41)):*
> ```
> Only the `setRoutes` has
> ```
Either **Only the `setRoutes` method has ...** or **Only `setRoutes` has ...**

---


*Comments from [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-router/136)*

**Comment by vlukashov** [comment]
Review status: 16 of 19 files reviewed at latest revision, 3 unresolved discussions.

---

*[demo/vaadin-router-route-parameters-demos.html, line 151 at r11](https://reviewable.io/reviews/vaadin/vaadin-router/136#-LEJ8Cid5k2X2ecaG3UN:-LEJ8Cid5k2X2ecaG3UO:b-g5jnbz) ([raw file](https://github.com/vaadin/vaadin-router/blob/ef801de58e91f4130634a5150fe753831ebbf226/demo/vaadin-router-route-parameters-demos.html#L151)):*
> ```
> Other characters would not meet the check and continue the route resolution
> ```
How about _Other characters would not meet the check and the route resolution would continue to other routes_?

---


*Comments from [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-router/136)*

**Comment by web-padawan** [comment]
Review status: 14 of 19 files reviewed at latest revision, 4 unresolved discussions.

---

*[demo/vaadin-router-route-parameters-demos.html, line 151 at r11](https://reviewable.io/reviews/vaadin/vaadin-router/136#-LEJ8Cid5k2X2ecaG3UN:-LEJ9gcz7i-c39yUVMaJ:b-896fix) ([raw file](https://github.com/vaadin/vaadin-router/blob/ef801de58e91f4130634a5150fe753831ebbf226/demo/vaadin-router-route-parameters-demos.html#L151)):*
<details><summary><i>Previously, vlukashov (Viktor Lukashov) wrote…</i></summary><blockquote>

> ```
> Other characters would not meet the check and continue the route resolution
> ```
How about _Other characters would not meet the check and the route resolution would continue to other routes_?
</blockquote></details>

Done.

---

*[src/router.js, line 182 at r5](https://reviewable.io/reviews/vaadin/vaadin-router/136#-LEJ6UGL27knqMRTJIZl:-LEJ8uZd8ttUkY2N2btJ:b-896fix) ([raw file](https://github.com/vaadin/vaadin-router/blob/87891e3454cf2f89d7216208a5a24c163af6e7c4/src/router.js#L182)):*
<details><summary><i>Previously, vlukashov (Viktor Lukashov) wrote…</i></summary><blockquote>

another closing back tick is missing
</blockquote></details>

Done.

---

*[src/router.js, line 30 at r8](https://reviewable.io/reviews/vaadin/vaadin-router/136#-LEJ72IQBaYqW7IZ3nyC:-LEJ8wlDCHtvio564yAH:b-896fix) ([raw file](https://github.com/vaadin/vaadin-router/blob/0f2435fb7cc69312f63cf60e669e68e26510be5f/src/router.js#L30)):*
<details><summary><i>Previously, vlukashov (Viktor Lukashov) wrote…</i></summary><blockquote>

> ```
> Use the `new Router(outlet)`
> ```

Use ~the~ `new Router(outlet)`
</blockquote></details>

Done.

---

*[src/router.js, line 41 at r8](https://reviewable.io/reviews/vaadin/vaadin-router/136#-LEJ7DyA9sAUkHhaijOV:-LEJ8w3O4ik6Goy8f2K0:b-896fix) ([raw file](https://github.com/vaadin/vaadin-router/blob/0f2435fb7cc69312f63cf60e669e68e26510be5f/src/router.js#L41)):*
<details><summary><i>Previously, vlukashov (Viktor Lukashov) wrote…</i></summary><blockquote>

> ```
> Only the `setRoutes` has
> ```
Either **Only the `setRoutes` method has ...** or **Only `setRoutes` has ...**
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-router/136)*
